shoppingflux/php-sdk
# #50 - http client adapter

### Link to the issue
https://github.com/shoppingflux/php-sdk/issues/50

### Reason for this PR
The goal is to remove the dependency with Guzzle6 client library to allow othe client to be used

### What does the PR do
- Remove dependecy to Guzzle library
- Add an `AdapterInterface` to allow other http client library to be used

### How to test
1. Install the SDK using the readme file of this PR
2. Call the SDK like before this PR, everything should work the same

You can use this test script to try on the SDK : https://gist.github.com/ddattee/71edb22ed8516ecd25e01d4fb71ad850

### Out of scope
Some code have been refactored to allow the code to be 100% tested
